test/e2e/apps/rc.go

```diff
@@ -558,3 +563,56 @@ func updateReplicationControllerWithRetries(c clientset.Interface, namespace, na
 	}
 	return rc, pollErr
 }
+
+type rcPatchFunc func(rc *v1.ReplicationController) ([]byte, error)
+
+func patchRCOrDie(cs kubernetes.Interface, dc dynamic.Interface, ns string, name string, f rcPatchFunc, subresources ...string) (updated *v1.ReplicationController) {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		old := getRCStatusOrDie(dc, ns, name)
+		patchBytes, err := f(old)
+		framework.ExpectNoError(err, "Failed to run ReplicationController patch function for resource %s", name)
+		if updated, err = cs.CoreV1().ReplicationControllers(ns).Patch(context.TODO(), old.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}, subresources...); err != nil {
+			return err
+		}
+		return nil
+	})
+
+	framework.ExpectNoError(err, "failed to patch the rc %q in namespace %q", name, ns)
+	waitForRCToBeProcessed(cs, ns, name)
+	return updated
+}
+
+func waitForRCToBeProcessed(cs kubernetes.Interface, ns string, name string) {
+	ginkgo.By("Waiting for the rc to be processed")
+	err := wait.PollImmediate(framework.Poll, schedulingTimeout, func() (bool, error) {
+		rc, err := cs.CoreV1().ReplicationControllers(ns).Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if rc.Status.ObservedGeneration < rc.Generation {
+			return false, fmt.Errorf("the ObservedGeneration for ReplicationController %s in namespaced %s is %v", name, ns, rc.Status.ObservedGeneration)
```

Comment:
It is better to ouput `rc.Generation` into the error message to investigate issues when this error happens.